### PR TITLE
Hw04

### PR DESCRIPTION
**Pull request by Vlad311010**

### CODE REVIEW

File: HW04/t2/Fraction.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+                num2 = Remainder;
+            }
+
+            return Math.Abs(num1);
+        }
+
+        public override string ToString() 
+        { 
+            return $"{_numerator}\\{_denominator}";
+        }
```
[the remark this answers is not in the document]
Comment (Vlad311010) [reply HW04/t2/Fraction.cs:51]: i don't see any benefits of using lambda expression.
simplified:
https://github.com/Vlad311010/homework/blob/8ddc5d061961c851820600dd76beaa8bf6e9788d/HW04/t2/Fraction.cs#L53-L57

File: HW04/t1/DiagonalMatrix.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            _elements = new T[size];
+        }
+
+        public DiagonalMatrix(params T[] elements)
+        {
+            if (elements == null)
+            {
+                elements = Array.Empty<T>();
+                return;
+            }
```
Comment (IgorKheidorov) [inline HW04/t1/DiagonalMatrix.cs:27]: Remove empty lines

File: HW04/t1/DiagonalMatrix.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        private bool IsDiagonalIndex(int i, int j)
+        {
+            return i == j;
+        }
+
+        private bool IsIndexValid(int i, int j)
+        {
+            return IsDiagonalIndex(i, j) && i >= 0 && i < _elements.Length;
+        }
+
```
Comment (IgorKheidorov) [inline HW04/t1/DiagonalMatrix.cs:69]: Remove line

File: HW04/t2/Fraction.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (IgorKheidorov) [reply HW04/t2/Fraction.cs:51]: Just shorter, not more. In this case lamba-style  provides more compact code, not more

**Review by IgorKheidorov — APPROVED** [review]
Ok

**Pull request merged**